contrib/download-frozen-image-v2.sh

```diff
@@ -201,6 +201,58 @@ handle_single_manifest_v2() {
 	manifestJsonEntries=("${manifestJsonEntries[@]}" "$manifestJsonEntry")
 }
 
+get_target_arch() {
+	if [ -n "${TARGETARCH}" ]; then
+		echo "${TARGETARCH}"
+		return 0
+	fi
+
+	if type go > /dev/null; then
+		echo "$(go env GOARCH)"
+		return 0
+	fi
+
+	if type dpkg > /dev/null; then
+		debArch="$(dpkg --print-architecture)"
+		case "${debArch}" in
+			armel | armhf)
+				echo "arm"
+				return 0
+				;;
+			*)
+				echo "${debArch}"
+				return 0
+				;;
+		esac
+	fi
+
+	if type uname > /dev/null; then
+		uArch="$(uname -m)"
+		case "${uArch}" in
+			x86_64)
+				echo amd64
+				return 0
+				;;
+			arm | armv6l | armv7l)
+				echo arm
+				return 0
+				;;
+			aarch64 | armv8l)
+				echo arm64
+				return 0
+				;;
+			*)
+				echo "${uArch}"
```

Comment:
This gets a little more complicated, unfortunately.  For example, on my `mips64le` machine, `uname -m` returns just `mips64` (even though in Go, that's big-endian MIPS64 which is incompatible... :facepalm:).

That being said, this is definitely fine for `s390x` and `ppc64le` (which appear to return the expected values).

Reply:
Fixed by above case

Reply:
Err... well "fixed"